File: app/src/main/java/com/deniscerri/ytdlnis/service/IDownloaderService.java
package com.deniscerri.ytdlnis.service;

import android.app.Activity;

import com.deniscerri.ytdlnis.database.models.ResultItem;

import java.util.ArrayList;

public interface IDownloaderService {
    DownloadInfo getInfo();
    void addActivity(Activity activity, ArrayList<IDownloaderListener> callback);
    void removeActivity(Activity activity);
    void updateQueue(ArrayList<ResultItem> queue);
    void cancelDownload(boolean cancelAll);
    void removeItemFromDownloadQueue(ResultItem video, String type);
}

File: app/src/main/java/com/deniscerri/ytdlnis/service/IDownloaderService.kt
package com.deniscerri.ytdlnis.service

import android.app.Activity
import com.deniscerri.ytdlnis.database.models.ResultItem

interface IDownloaderService {
    val info: DownloadInfo?
    fun addActivity(activity: Activity?, callback: ArrayList<IDownloaderListener?>?)
    fun removeActivity(activity: Activity?)
    fun updateQueue(queue: ArrayList<ResultItem?>?)
    fun cancelDownload(cancelAll: Boolean)
    fun removeItemFromDownloadQueue(video: ResultItem?, type: String?)
}